server/src/main/java/com/ncc/neon/better/IENlpModule.java

```diff
@@ -0,0 +1,67 @@
+package com.ncc.neon.better;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ncc.neon.services.BetterFileService;
+import com.ncc.neon.services.DatasetService;
+import com.ncc.neon.services.FileShareService;
+import org.elasticsearch.rest.RestStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Flux;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Map;
+
+@Component
+public class IENlpModule extends NlpModule {
+    private HttpEndpoint trainEndpoint;
+    private HttpEndpoint trainListEndpoint;
+    private HttpEndpoint infEndpoint;
+    private HttpEndpoint infListEndpoint;
+
+    @Autowired
+    IENlpModule(DatasetService datasetService, FileShareService fileShareService, BetterFileService betterFileService) {
+        super(datasetService, fileShareService, betterFileService);
+    }
+
+    @Override
+    protected void setEndpoints(HttpEndpoint[] endpoints) {
+        for (HttpEndpoint endpoint : endpoints) {
+            switch (endpoint.getType()) {
+                case TRAIN:
+                    trainEndpoint = endpoint;
+                    break;
+                case TRAIN_LIST:
+                    trainListEndpoint = endpoint;
+                    break;
+                case INF:
+                    infEndpoint = endpoint;
+                    break;
+                case INF_LIST:
+                    infListEndpoint = endpoint;
+                    break;
+            }
+        }
+    }
+
+    public Flux<RestStatus> performTraining(String listConfigFile, String trainConfigFile) throws IOException {
+        // Parse JSON files to maps.
+        String shareDir = System.getenv("SHARE_DIR");
```

Comment:
Should this be a variable in the super class instead of instantiating it every time this function is called?

Reply:
Good catch